### PR TITLE
Update CAMARA-API-access-and-user-consent.md

### LINKED ISSUE #67: Make the technical ruleset description precise.
**Problem description**
The technical ruleset in user consent should state that it gets applied only after a subproject decides to adopt a 3-legged flow. Also, that it is a recommendation which will promote alignment.

**Expected behavior**

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by shilpa-padgaonkar**
Made changes as per https://github.com/camaraproject/IdentityAndConsentManagement/issues/57#issuecomment-1733917140 and https://github.com/camaraproject/IdentityAndConsentManagement/issues/57#issuecomment-1734278680

#### What type of PR is this?

Add one of the following kinds:
* correction

#### Which issue(s) this PR fixes:

Fixes #67 

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by jpengar** [comment]
@shilpa-padgaonkar Hi Shilpa, did you get to see my comments above?

**Comment by shilpa-padgaonkar** [comment]
@jpengar Actually I dont see any comments from you in this PR. Have you added them in some issue or could you kindly paste your comment links here?

**Comment by jpengar** [comment]
> @jpengar Hi Jesus, I don't see any comments from you in this PR. Have you added them in some issue or could you kindly paste your comment links here?

Do you see it now? I think I didn't submit my review (facepalm). My bad.

**Comment by shilpa-padgaonkar** [comment]
Yes, now I see it :)
It was not just moving the lines at the end and bringing them together under the section technical ruleset, but it also took the more flexible/lenient approach by changing from "must" to "should".

I am fine with repeating the line twice and keeping it under different sections, if that's the way you would prefer it in the doc (wouldn't mind one way over the other). Will update the PR

**Comment by shilpa-padgaonkar** [comment]
@jpengar  have made the changes now. Please take a look.
I have explicitly kept this part of your suggestion out "Every time personal user data is processed by an API and the user can exercise their rights either via opt-in and/or opt-out, 3-legged access tokens must be used." as it will again trigger this legal related discussions and unnecessarily block this PR.

**Comment by shilpa-padgaonkar** [comment]
@jpengar  have made the changes now. Please take a look.
I have explicitly kept this part of your suggestion out "Every time personal user data is processed by an API and the user can exercise their rights either via opt-in and/or opt-out, 3-legged access tokens must be used." as it will again trigger this legal related discussions and unnecessarily block this PR.

**Comment by shilpa-padgaonkar** [comment]
@jpengar Could we discuss it at TSC meeting venue? I might have to miss the WG meeting on Wednesday as I am travelling. I don't mind keeping your line in, as long as we mention in the line that it also depends on legal decisions.

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (rartych) [inline documentation/CAMARA-API-access-and-user-consent.md:174]: Small typo:
```suggestion
If all API usecases point to the need of On-net scenario and where the consumption device and authentication device are the same, the front end flow should be used. eg. NumberVerification
```

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (rartych) [inline documentation/CAMARA-API-access-and-user-consent.md:305]: Unification of casing for flow names:

```suggestion
The technical ruleset is applicable only after a subproject has agreed to use a 3-legged authentication flow. This ruleset provides a recommendation which will help API providers to align on the 3-legged flow and help with aggregation. If some usecase/s for an API point to off-net scenarios and where consumption and authenticaiton devices could be different, the Backend flow should be used.
```

**Review by jpengar — COMMENTED** [review]
>@jpengar Could we discuss it at TSC meeting venue? I might have to miss the WG meeting on Wednesday as I am travelling. I don't mind keeping your line in, as long as we mention in the line that it also depends on legal decisions.

I've reviewed the PR again and have included the previously suggested clarifying sentence in the FE flow ruleset and the BE flow ruleset. Feel free to add further clarification on legal decision dependency. Otherwise we can merge this as is if there are no further objections.

**Comment by shilpa-padgaonkar** [comment]
LGTM

Not related to this PR, but the doc still stays that issue purpose is still under discussion. Should this line be removed now?

**Comment by jpengar** [comment]
>LGTM
Not related to this PR, but the doc still stays that issue purpose is still under discussion. Should this line be removed now?

Good. Yes, we can remove that line as well. Actually, I didn't make any changes just using the github suggestion feature, because you were the owner of this PR. So feel free to consolidate those changes. Or do you want me to do it?

**Comment by shilpa-padgaonkar** [comment]
I see that the issue32 is still kept open (do you know why?). I guess, best way would be to open a new PR for that change (don't want that someone complains that purpose line change is not related to this PR etc. ).
Also, we can go ahead and merge this one.

**Comment by shilpa-padgaonkar** [comment]
(facepalm) I didnt realise that you didnt commit the suggested change. Hence i directly only viwed the file :/ Will take a look now

**Comment by jpengar** [comment]
Actually issue #32 was pending to close and I've closed it already.

**Comment by shilpa-padgaonkar** [comment]
@jpengar Done.

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:174]: @shilpa-padgaonkar I think you should apply this commit as well to correct the typos detected by @rartych

File: documentation/CAMARA-API-access-and-user-consent.md
Comment (jpengar) [inline documentation/CAMARA-API-access-and-user-consent.md:176]: Duplicated.

**Review by jpengar — APPROVED** [review]
@shilpa-padgaonkar LGTM in advance. You just need to correct a couple of typos. Thanks!

**Comment by jpengar** [comment]
@shilpa-padgaonkar Could you have a look at my last two comments. I think we are good to merge this.

**Comment by jpengar** [comment]
@shilpa-padgaonkar That's fine now. Could you also remove this line at the end of the section? I think it no longer applies and that issue is closed.

`>[TO BE EDITED/COMPLETED] - [Technical disscussion on ruleset to be applied](https://github.com/camaraproject/IdentityAndConsentManagement/issues/52) are ongoing.`

I tried to do it myself so as not to ask you, but I can't fetch your branch to do so :(

I will then give my approval and you can MERGE it.

**Review by jpengar — APPROVED** [review]
LGTM

**Pull request merged**
